### PR TITLE
Add Segments Explainer 

### PR DESCRIPTION
**Pull request by ststimac**
Since the thinking for window segments for multi-screen devices evolved to be a `segments` property hanging off the visual viewport API, we're archiving the windows segments explainer and will point to this segments explainer that has been updated with the latest thinking, to be included here in the Visual Viewport repo.

This PR: 
Adds the Segments explainer to the repo
Fixes a typo in the draft spec

@dlibby-

### CODE REVIEW

**Comment by ststimac** [comment]
I'm having issues revalidating after linking my W3C account.

**Comment by scottlow** [comment]
@dlibby- @bokand, any feedback on this one before merging? It'd be great to land this so we can point the TAG to it for final review. Thanks!

File: segments-explainer/SEGMENTS-EXPLAINER.md
Comment (dlibby-) [inline segments-explainer/SEGMENTS-EXPLAINER.md]: This should be updated to handle the null segments case instead of testing for length > 1

File: segments-explainer/SEGMENTS-EXPLAINER.md
Comment (dlibby-) [inline segments-explainer/SEGMENTS-EXPLAINER.md:82]: Can we link to the CSS specs (and/or give some examples) to tie these concepts together?

File: segments-explainer/SEGMENTS-EXPLAINER.md
Comment (dlibby-) [inline segments-explainer/SEGMENTS-EXPLAINER.md]: instead of 'on a desktop' I think we should say when there is only a single segment, and mention it is not exposed in this case because this information would be redundant with other fields on VisualViewport (and also could lead to future compat issues on these devices if authors started using visualViewport.segments[0] to target single-screen devices)

**Comment by bokand** [comment]
Sorry for the delay - I just got back from a leave.

I'm happy to merge this when @dlibby is happy with it - let me know.

File: segments-explainer/SEGMENTS-EXPLAINER.md
Comment (dlibby-) [inline segments-explainer/SEGMENTS-EXPLAINER.md]: Sorry for missing this before, but this needs the same treatment as the previous script block. Also, we probably want to have something like:

```js  
window.onresize = function() {
    const segments = window.visualViewport.segments;
    if (segments && segments.length > 1) {
        // Make changes two split content into the segments.
    } else {
        // Reset state to single viewport (normal responsive layout).
    }
}
```

**Review by dlibby- — CHANGES_REQUESTED** [review]
LGTM with one remaining suggestion.

**Comment by ststimac** [comment]
Made @dlibby-'s update, so should be good to go now. Thanks!

**Review by dlibby- — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
